networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/gossip/AttestationSubnetSubscriptions.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.networking.eth2.gossip;
+
+import static java.lang.StrictMath.toIntExact;
+import static tech.pegasys.artemis.util.config.Constants.ATTESTATION_SUBNET_COUNT;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import io.vertx.core.impl.ConcurrentHashSet;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import tech.pegasys.artemis.networking.eth2.gossip.topics.AttestationTopicHandler;
+import tech.pegasys.artemis.networking.p2p.gossip.GossipNetwork;
+import tech.pegasys.artemis.networking.p2p.gossip.TopicChannel;
+import tech.pegasys.artemis.storage.client.RecentChainData;
+
+public class AttestationSubnetSubscriptions implements AutoCloseable {
+  private final GossipNetwork gossipNetwork;
+  private final RecentChainData recentChainData;
+  private final EventBus eventBus;
+
+  private final Map<Integer, Set<UnsignedLong>> subnetIdToCommittees = new ConcurrentHashMap<>();
+  private final Map<Integer, TopicChannel> subnetIdToTopicChannel = new ConcurrentHashMap<>();
+
+  public AttestationSubnetSubscriptions(
+      final GossipNetwork gossipNetwork,
+      final RecentChainData recentChainData,
+      final EventBus eventBus) {
+    this.gossipNetwork = gossipNetwork;
+    this.recentChainData = recentChainData;
+    this.eventBus = eventBus;
+  }
+
+  public Optional<TopicChannel> getChannel(final UnsignedLong committeeIndex) {
+    final int subnetId = committeeIndexToSubnetId(committeeIndex);
+    if (!subnetIdToCommittees
+        .computeIfAbsent(subnetId, __ -> Collections.emptySet())
+        .contains(committeeIndex)) {
+      // We're not subscribed to this committee subnet
+      return Optional.empty();
+    }
+    return Optional.ofNullable(subnetIdToTopicChannel.get(subnetId));
+  }
+
+  public synchronized void subscribeToCommitteeTopic(final UnsignedLong committeeIndex) {
```

Comment:
We should pick either using `synchronized` or using concurrent collections rather than mixing both.  With the current approach of using two maps I think we'd need `synchronized` so probably just make them all a normal `HashMap` and `HashSet` and add `synchronized` to `getChannel` and `close`.

Reply:
Makes sense. Done.